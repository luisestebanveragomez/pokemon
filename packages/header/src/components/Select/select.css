.select__ul {
    list-style: none;
    background: rgb(255 255 255 / 37%);
    border-radius: .5rem;
    overflow: hidden;
}

.select__ul-list {
    transition: .3s;
    background: transparent;
}

.select__ul-list:hover {
    transition: .3s;
    background: rgb(255 255 255 / 37%);
}

.select__ul-button {
    display: flex;
    padding: .3rem .5rem .3rem 0;
    background: transparent;
    width: 100%;
    color: var(--white);
    cursor: pointer;
    align-items: center;
}

.select__ul-button span {
    flex-basis: 30px;
    font-size: .6rem;
    display: flex;
    align-items: center;
    justify-content: center;
}