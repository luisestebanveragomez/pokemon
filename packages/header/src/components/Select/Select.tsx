import { FC } from "react";

import "./select.css";

interface SelectProps {
  languages: string[];
  active: string;
  handle: (item: string) => void;
}

export const Select: FC<SelectProps> = ({ languages, active, handle }) => {
  return (
    <ul className="select__ul">
      {languages.map((item) => (
        <li key={item} className="select__ul-list">
          <button className="select__ul-button" onClick={() => handle(item)}>
            <span>{active === item && <i className="icon-checkmark"></i>}</span>
            {item}
          </button>
        </li>
      ))}
    </ul>
  );
};
