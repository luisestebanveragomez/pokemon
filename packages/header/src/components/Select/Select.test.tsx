import { fireEvent, render, screen } from "@testing-library/react";
import { Select } from "./Select";

describe("Select", () => {
  it("Should Render", () => {
    let { container } = render(
      <Select languages={[]} active={""} handle={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Should has two Elements", () => {
    let { container } = render(
      <Select languages={["es", "en"]} active={"en"} handle={() => {}} />
    );
    const lists = container.querySelectorAll("li");
    expect(lists.length).toBe(2);
  });

  it("Should be return value", () => {
    let lan = "";
    const changeLan = (el: string) => (lan = el);
    let { container } = render(
      <Select languages={["es", "en"]} active={"en"} handle={changeLan} />
    );
    const secondButton = container.querySelectorAll("button")[1];
    fireEvent.click(secondButton);
    expect(lan).toBe("en");
  });
});
