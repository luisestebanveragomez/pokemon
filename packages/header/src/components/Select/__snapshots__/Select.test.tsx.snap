// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should Render 1`] = `
<div>
  <ul
    class="select__ul"
  />
</div>
`;
